task/aws/resources/data_source_bucket_test.go

```diff
@@ -0,0 +1,58 @@
+package resources_test
```

Comment:
Tests are nice, although mocking cloud infrastructure is rather risky. Maybe we can rely on `rclone` to use functionality that it's already tested on their side or can be tested locally with the same API as cloud resources?

Reply:
I don't see how mocking cloud apis (not infrastructure) is riskier than having no tests at all.

My current proposal is twofold:
1. Each resource defines limited interfaces (implemented by corresponding cloud apis) that include only the methods actually used.
2. We generate mocks for those interfaces and use them in tests to ensure conditions are handled correctly and api calls are made in the right order.

Reply:
If your proposal implies accounting for a considerable amount of cloud-specific details (e.g. “limited interfaces” for S3 and EC2, others for Cloud Storage and Compute Engine, ...) I wonder if maintaining them will be any easier than testing the code against actual cloud APIs, even if it's noticeably slower.